### ISSUE TITLE
Fix S3253 FP: Conflicts with CS8983 - A 'struct' with field initializers must include an explicitly declared constructor.

### ISSUE THREAD

**Issue by rjgotten** [SonarSource/sonar-dotnet#8087]
### Description

S3253 is raised on empty constructors that must be added to readonly structs containing field initializers, as per [CS8983](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/constructor-errors?f1url=%3FappId%3Droslyn%26k%3Dk(CS8983)#constructors-in-struct-types).

### Repro steps

```cs
public readonly struct Example
{
  public Example() { }
  
  public bool Foo { get; init; } = true;
}
```

### Expected behavior

S3253 is not raised, because the constructor is actually required.

### Actual behavior

S3253 is raised.

### Known workarounds

None. Suppress the rule.

### Related information

* C#/VB.NET Plugins version : C# 9.8.0.76515 (SonarLint  7.3.0.77872)
* Visual Studio version: 17.7.4
* MSBuild / dotnet version: dotnet 7.0.401
* Operating System: Windows 11

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @rjgotten! I confirm this to be an FP indeed, your issue has been added to our backlog and we will try to tackle it in a future sprint.
Thanks for reporting this!
